sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core.Pipeline.Policies;
+
+namespace Azure.Core.Pipeline
+{
+    public class HttpPipelineBuilder
+    {
+        private readonly HttpClientOptions _options;
+        private readonly List<PolicyHolder> _policies;
+
+        public HttpPipelineBuilder(HttpClientOptions options)
+        {
+            _options = options;
+            _policies = new List<PolicyHolder>(5)
+            {
+                new PolicyHolder(HttpClientOptions.TelemetryPolicy, new TelemetryPolicy(options.Telemetry, options.GetType().Assembly)),
+                new PolicyHolder(HttpClientOptions.ClientRequestIdPolicy, ClientRequestIdPolicy.Singleton),
+                new PolicyHolder(HttpClientOptions.RetryPolicy, null),
+                new PolicyHolder(HttpClientOptions.LoggingPolicy, LoggingPolicy.Shared),
+                new PolicyHolder(HttpClientOptions.TransportPolicy, new HttpPipelineTransportPolicy(options.Transport))
+            };
+        }
+
+        public void InsertBefore(string beforePolicy, string policyName, HttpPipelinePolicy policy)
```

Comment:
Can we make it such that there are two ways to configure the builder:
a) by passing before and after retry
b) very advanced index based (optional feature, i.e. I don't care about for now, but it might be good to think how we would add it if we need to).

Reply:
Done